### PR TITLE
Changed security on the individual_admins endpoint

### PR DESCRIPTION
**Pull request by RishiDiwanTT**
## Description
Only Library Managers and System Admins may access this endpoint
Library Users will receive a 403

This PR relates to 2 tickets
- [View restrictions](https://www.notion.so/lyrasis/CM-Remove-ability-for-library-Users-to-view-all-admins-in-a-Collection-Manager-ea01a9e96bd94fe08ca0d081e062aee6)
- [Edit restrictions](https://www.notion.so/lyrasis/Remove-ability-for-library-staff-administrator-role-to-see-and-edit-other-library-s-login-informatio-dd416398e16443e3aa675f563d7955b2)

The [view ticket](https://www.notion.so/lyrasis/CM-Remove-ability-for-library-Users-to-view-all-admins-in-a-Collection-Manager-ea01a9e96bd94fe08ca0d081e062aee6) has been implemented as such
System admins - Can view everything
Sitewide Managers - Can view Sitewide managers and Library managers
Library Managers - Can view Sitewide managers (as they are also managing their library) and managers of the same libraries

The [edit ticket](https://www.notion.so/lyrasis/Remove-ability-for-library-staff-administrator-role-to-see-and-edit-other-library-s-login-informatio-dd416398e16443e3aa675f563d7955b2) mentions the need to stop managers from being able to edit the details of other users, however while testing it seems this is already in place.
Editing a profile is limited to either System Admins, Sitewide Managers, or Managers of the same library.

## Motivation and Context
Only administrators should not be able to view Administrators of any library

## How Has This Been Tested?

The previous tests were changed to cover this scenario

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by RishiDiwanTT** [comment]
Converted to draft so we can add [the second security update](https://www.notion.so/lyrasis/Remove-ability-for-library-staff-administrator-role-to-see-and-edit-other-library-s-login-informatio-dd416398e16443e3aa675f563d7955b2) to the same PR.
They are dependant changes

File: api/admin/controller/individual_admin_settings.py [hunk 1]
```diff
@@ -19,20 +20,78 @@ def process_individual_admins(self):
...
             return self.process_post()
 
+    def _highest_authorized_role(self) -> Optional[AdminRole]:
+        highest_role = None
+        has_auth = False
+
+        admin = getattr(flask.request, "admin", None)
+
+        if not admin:
+            return False
```
[the remark this answers is not in the document]
Comment (RishiDiwanTT) [reply api/admin/controller/individual_admin_settings.py]: What... was I thinking? And how did mypy allow this?

**Comment by io7m** [comment]
Oops, I see you already fixed it. :laughing:

**Review by io7m — APPROVED** [review]
Looks good!

**Pull request merged**